### PR TITLE
[FIX] web: search input date in another timezone

### PR DESCRIPTION
**Pull request by kebeclibre**
Put the browser in timezone Sao Paulo (Or like West of UTC)

Trigger the autocomplete (and the subsequent search) on a date field
by writing a date (in the locale format) in the search view

Before this commit, the domain sent to the server contained the date as the day
**before** the one asked for in the input

This is because, the string input is parsed and gives:
input = 12/02/2018
When creating the moment object:
> since there is no explicit time, moment will interpret it as 12am (midnight)
> We force moment to consider the string as being UTC (function: moment.utc())
> the moment contains, as output the time ** 2018-02-12T00:00:00 UTC **

When getting the facet value for making the domain
> we call toDate on the moment object, which, according to the browser is UTC and will
be converted to the locale timezone before formatting
> And the domain will use the date 2018-02-11T22:00:00 Brazil/SaoPaulo
which gives the day before the one we asked for

After this commit, this issue doesn't arise, because we don't convert the moment into a Date,
and we directly format that moment into a string

OPW 1903224

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/views/search/search_inputs.js [hunk 1]
```diff
@@ -346,7 +346,8 @@ var BooleanField = SelectionField.extend(/** @lends instance.web.search.BooleanF
  */
 var DateField = Field.extend(/** @lends instance.web.search.DateField# */{
     value_from: function (facetValue) {
-        return time.date_to_str(facetValue.get('value'));
+        var serverFormat = (this.attrs && this.attrs.type === 'datetime') ? 'YYYY-MM-DD HH:mm:ss' : 'YYYY-MM-DD';
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/search/search_inputs.js]: it's weird to check if it is a datetime in the DateField widget, whereas there is a specific DateTimeField one

File: addons/web/static/tests/views/search_view_tests.js [hunk 2]
```diff
@@ -461,6 +461,95 @@ QUnit.module('Search View', {
 
     QUnit.module('Filters Menu');
 
+    QUnit.test('Search date and datetime fields. Support of timezones', function (assert) {
+        assert.expect(4);
+
+        this.data.partner.fields.birth_moment = {string: "Birth Moment", type: "datetime", store: true, sortable: true};
```
Comment (aab-odoo) [inline addons/web/static/tests/views/search_view_tests.js]: i'd call it birth_datetime or something, to emphasize that aspect

File: addons/web/static/src/js/views/search/search_inputs.js (hunk 1 above)
Comment (kebeclibre) [reply addons/web/static/src/js/views/search/search_inputs.js]: @aab-odoo 
Yes it is
I can do the choice of the format of date/time in the init(), either way that check would have to be done in the super Class DateField

**Comment by kebeclibre** [comment]
@aab-odoo 
Changed variable name in test

@Polymorphe57 
Ready for review

**Comment by kebeclibre** [comment]
robodoo r+

**Pull request closed without merging**